Add uncrustify config

thenor opened the pull request:
Create a basic uncrustify config which matches the C++ formatting guidelines.
Create a script that executes uncrustify on the cpp, h and c files in the main code dirs.

Signed-off-by: Therese Nordqvist <therese.nordqvist@pelagicore.com>

breakreturn commented on CMakeLists.txt:
```diff
@@ -304,6 +307,10 @@ if(ENABLE_DBUSGATEWAY)
 endif()
 
+if(ENABLE_UNCRUSTIFY)
+    ## TODO: change this to use a config source
```
Is this left intentionally? What does it mean?

thenor replied:
This was left intentionally since I didn't start using this flag. I could remove the change completely since it doesn't do anything at the moment.

breakreturn commented on servicetest/timingprofiling/simple.c:
```diff
@@ -21,6 +20,7 @@
 #include <unistd.h>
 
-int main(int argc, char **argv) {
+int main(int argc, char * *argv)
```
This should not be a part of this commit right?

Also, if this is how the config wants the code to look, could we configure that so it wants it to look like we want it?

thenor replied:
It was included by mistake. I'm removing it from the commit.

Commit:
Add uncrustify config

Create a basic uncrustify config which matches the C++ formatting
guidelines.
Create a script that executes uncrustify on the cpp, h and c files
in the main code dirs.

Signed-off-by: Therese Nordqvist <therese.nordqvist@pelagicore.com>

frznlogic commented on scripts/format_sources.sh:
```diff
@@ -0,0 +1,73 @@
+#
+
+#
+# This is a helper script for invoking uncrustify on a set of C++ files.
```
These comments are wrong. The output is stored in a prefix

thenor replied:
Oops copy paste error. Thanks for catching it.

frznlogic replied:
Np:)

frznlogic commented on scripts/format_sources.sh:
```diff
@@ -0,0 +1,73 @@
+
+read -p "We are going to format the C/C++ (*.cpp, *.c, *.h) source files from the folder
+'$COMPONENT_FOLDER'. We recommend that you first commit your files to your local GIT repo
+before going further, since the formatter will modify your files. You will be able to amend
```
This is also not true... we are not going to overwrite anything as far as I can tell.

thenor replied:
Same as above. I forgot to fix the comment.
Also: when we change to use this with make or in the CI system we probably don't want this prompt, since it waits for an answer.

thenor replied:
Should I remove it completely, perhaps?

frznlogic replied:
I think so... it simplifies using the script automatically if nothing else.

frznlogic commented on scripts/format_sources.sh:
```diff
@@ -0,0 +1,73 @@
+#
+
+COMPONENT_FOLDER="" # Set to softwarecontainer base dir
+UNCRUSTIFY_CONFIG_FILE="" # Add option to set this
```
Do we want to add support for reading this from command line?

thenor replied:
I had some idea that it would use the default config file unless specified, but that another config could be used with a flag/argument to "make uncrustify".

frznlogic commented:
The comments and the read -p command are wrong with regards to where data is stored.

frznlogic commented on scripts/format_sources.sh:
```diff
@@ -0,0 +1,63 @@
+# uncrustify config file to use.
+#
+
+UNCRUSTIFY_LOGFILE="uncrustify_output.log" # Currently not used
```
Could you make the same pattern as the other variables for this one?

thenor replied:
Of course!
